### ISSUE TITLE
Automate elasticsearch deployment

### ISSUE THREAD

**Issue by pameck** [snitchhunt/automation#1]
We want to be able to deploy an instance of  [AWS ES](https://aws.amazon.com/elasticsearch-service/) by running a script.

- [ ] Create a terraform script to deploy an instance of AWS ES
- [ ] The Terraform state should be remote. see [this](https://www.terraform.io/docs/state/remote.html)
- [ ] No secrets committed to the repo
- [ ] Use the most recent version of Elastic Search supported by AWS ES.
- [ ] Determine what is a reasonable configuration for the cluster (re: instance type, # of instances)
- [ ] Determine what is a reasonable configuration for the storage
- [ ] Elastic search should not be public accessible

**Comment by pameck** [comment]
We could use the [AWS ES](https://aws.amazon.com/elasticsearch-service/) and make it easier for us @rdoh @akjones ?

**Comment by rdoh** [comment]
Sounds like a good plan to me @pameck. Either that or elastic.co's hosted service. Just depends on costs, and our ability to automate instantiation of those services.

**Comment by pameck** [comment]
Looking at  **elastic.co**, subscriptions start at $45/month + resources used. Seems a bit pricey.  
https://www.elastic.co/cloud/as-a-service/subscriptions

AWS is like everything with AWS pricing, an enigma. (at least is per hour, and no subscription fees)
https://aws.amazon.com/elasticsearch-service/pricing/

**Comment by pameck** [comment]
AWS ES has a built-in integration with Kibana, which means [issue#9](https://github.com/snitchhunt/automation/issues/9) is automatically done if we use AWS hosted solution.

**Comment by akjones** [comment]
AWS ES is looking good. Do we know what version they offer?

**Comment by pameck** [comment]
@akjones according to the docs: 

>Amazon Elasticsearch Service currently supports Elasticsearch versions 5.3, 5.1, 2.3, and 1.5.

**Comment by akjones** [comment]
Excellent. Old snitch hunt needed version 4 to run, but I reckon we can
make it work in 5.3.

**Comment by pameck** [comment]
After having a chat with @rshamsundar, @rdoh, @AliciaSJ and Eugene, it seems we will be using AWS ES for this. I will be changing the story description and adding some tasks so we can capture the details of the work we need to do.

**Comment by c4milo** [comment]
Just make sure you are aware of AWS quirks before committing to it: https://read.acloud.guru/things-you-should-know-before-using-awss-elasticsearch-service-7cd70c9afb4f

**Comment by pameck** [comment]
Thanks @c4milo, I'll check that out! :)

**Comment by rdoh** [comment]
I don't think we can use AWS because their service doesn't provide the Graph plugin (which is needed for the last challenge). It's also very difficult to lock down access because AWS doesn't allow you to use Kibana's security management - you have to use a policy to lock down by IP.

Elastic Cloud does have the Graph plugin fwiw (and allows us to create Kibana users), so we could use that, or update the old build-elasticsearch scripts for configuring a VM. The latter is probably cheaper for our users. I'm not sure what is easier to automate.
